Framework/DataHandling/src/LoadEMU.cpp

```diff
@@ -289,6 +294,29 @@ double maskedStdev(const std::vector<double> &vec, const std::vector<bool> &mask
   return std::sqrt(sum / static_cast<double>(count));
 }
 
+// calculate a running average for a given half window size assuming the data is wrapped
+std::vector<double> runningAverage(const std::vector<size_t> &data, size_t halfWindow) {
+  const auto N = data.size();
+  const size_t TN = 2 * halfWindow + 1;
```

Comment:
I found it a bit hard to follow the logic in this function. Is it worth added some more comments and/or descriptive variable names?

Reply:
I've added a detailed explanation of the filter to the code base and changed the variable name.